src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py

```diff
@@ -326,3 +307,36 @@ def miscs_update_idxs_vals(self, miscs, idxs, vals,
                 if assert_all_vals_used or tid in misc_by_id:
                     misc_by_id[tid]['idxs'][key] = [tid]
                     misc_by_id[tid]['vals'][key] = [val]
+
+    def get_suggestion(self, random_search=False):
+        """get suggestion from hyperopt
+
+        Keyword Arguments:
+            random_search {bool} -- [flag to indicate random search or not] (default: {False})
+
+        Returns:
+            [dict] -- [parameter suggestion]
+        """
+
+        rval = self.rval
+        trials = rval.trials
+        algorithm = rval.algo
+        new_ids = rval.trials.new_trial_ids(1)
+        rval.trials.refresh()
+        random_state = rval.rstate.randint(2**31-1)
+        if random_search:
+            new_trials = hp.rand.suggest(new_ids, rval.domain, trials, random_state)
+        else:
+            new_trials = algorithm(new_ids, rval.domain, trials, random_state)
+        rval.trials.refresh()
+        vals = new_trials[0]['misc']['vals']
+        parameter = dict()
+        for key in vals:
+            try:
+                parameter[key] = vals[key][0].item()
+            except Exception:
```

Comment:
please try to catch specific Exception here.

Reply:
OK